test/logstream/kubelogs.go

```diff
@@ -49,54 +53,99 @@ var _ streamer = (*kubelogs)(nil)
 // timeFormat defines a simple timestamp with millisecond granularity
 const timeFormat = "15:04:05.000"
 
+func (k *kubelogs) startForPod(eg *errgroup.Group, pod *corev1.Pod) {
+	k.watchedPods.Insert(pod.Name)
+	// Grab data from all containers in the pods.  We need this in case
+	// an envoy sidecar is injected for mesh installs.  This should be
+	// equivalent to --all-containers.
+	for _, container := range pod.Spec.Containers {
+		// Required for capture below.
+		pod, container := pod, container
```

Comment:
We really only need to capture `pod.Name` and `container.Name`, right?

Reply:
and pod namespace. I can reduce this to `pns, pn, cn :=...`, given than containers are copied as a whole.